pkg/handlers/authentication/okta_auth_code_flow.go

```diff
@@ -0,0 +1,125 @@
+package authentication
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	verifier "github.com/okta/okta-jwt-verifier-golang"
+	"go.uber.org/zap"
+
+	"github.com/transcom/mymove/pkg/appcontext"
+	"github.com/transcom/mymove/pkg/handlers/authentication/okta"
+	"github.com/transcom/mymove/pkg/models"
+)
+
+// ! See flow here:
+// ! https://developer.okta.com/docs/guides/implement-grant-type/authcode/main/
+
+func getProfileData(appCtx appcontext.AppContext, provider okta.Provider) (models.OktaUser, error) {
+	user := models.OktaUser{}
+
+	if appCtx.Session().AccessToken == "" {
+		return user, nil
+	}
+
+	reqURL := provider.GetUserInfoURL()
+
+	req, _ := http.NewRequest("GET", reqURL, bytes.NewReader([]byte("")))
+	h := req.Header
+	h.Add("Authorization", "Bearer "+appCtx.Session().AccessToken)
+	h.Add("Accept", "application/json")
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		appCtx.Logger().Error("could not execute request", zap.Error(err))
+	}
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		appCtx.Logger().Error("could not read response body", zap.Error(err))
+	}
+	defer resp.Body.Close()
```

Comment:
Minor issue: I think it would make sense to return errors in both of these `if` statements. If we continue executing after hitting the first error, it will most likely just hit all of the other error handlers, which will result in more errors in the logs, which will make them just a little harder to read.

```suggestion
	if err != nil {
		appCtx.Logger().Error("could not execute request", zap.Error(err))
		return models.OktaUser{}, err
	}
	body, err := io.ReadAll(resp.Body)
	defer resp.Body.Close()
	if err != nil {
		appCtx.Logger().Error("could not read response body", zap.Error(err))
		return models.OktaUser{}, err
	}
```